### ISSUE TITLE
[Bug]: Not able to use Gemini.

### ISSUE THREAD

**Issue by ujjwalk906** [run-llama/llama_index#12100]
### Bug Description

Getting key error when calling Gemini chat.

KeyError: 'model'

### Version

0.10.20

### Steps to Reproduce

from llama_index.llms.gemini import Gemini
from llama_index.core.llms import ChatMessage, MessageRole

history = [
    ChatMessage(role=MessageRole.USER,
                content= " The alien says hello ")
]

llm = Gemini(api_key=credentials)
llm.chat(history)

### Relevant Logs/Tracbacks

```shell
{
	"name": "KeyError",
	"message": "'model'",
	"stack": "---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[26], line 10
      4 history = [
      5     ChatMessage(role=MessageRole.USER,
      6                 content= \" The alien says hello \")
      7 ]
      9 llm = Gemini(api_key=credentials)
---> 10 llm.chat(history)

File d:\\Anaconda\\envs\\llama_0_10\\Lib\\site-packages\\llama_index\\core\\llms\\callbacks.py:93, in wrapped_llm_chat(_self, messages, **kwargs)
     83         callback_manager.on_event_end(
     84             CBEventType.LLM,
     85             payload={
   (...)
     89             event_id=event_id,
     90         )
     92     return wrapped_gen()
---> 93 else:
     94     callback_manager.on_event_end(
     95         CBEventType.LLM,
     96         payload={
   (...)
    100         event_id=event_id,
    101     )
    102     dispatcher.event(
    103         LLMChatEndEvent(
    104             messages=messages,
    105             response=f_return_val,
    106         )
    107     )

File d:\\Anaconda\\envs\\llama_0_10\\Lib\\site-packages\\llama_index\\llms\\gemini\\base.py:163, in Gemini.chat(self, messages, **kwargs)
    161 chat = self._model.start_chat(history=history)
    162 response = chat.send_message(next_msg)
--> 163 return chat_from_gemini_response(response)

File d:\\Anaconda\\envs\\llama_0_10\\Lib\\site-packages\\llama_index\\llms\\gemini\\utils.py:61, in chat_from_gemini_response(response)
     55 _error_if_finished_early(top_candidate)
     57 raw = {
     58     **(type(top_candidate).to_dict(top_candidate)),
     59     **(type(response.prompt_feedback).to_dict(response.prompt_feedback)),
     60 }
---> 61 role = ROLES_FROM_GEMINI[top_candidate.content.role]
     62 return ChatResponse(message=ChatMessage(role=role, content=response.text), raw=raw)

KeyError: 'model'"
}
```

**Comment by logan-markewich** [comment]
Try `pip install -U llama-index-core llama-index-llms-gemini`

**Comment by ujjwalk906** [comment]
> Try `pip install -U llama-index-core llama-index-llms-gemini`

Still the same issue persists.

**Comment by logan-markewich** [comment]
I don't really see how that's possible?

https://github.com/run-llama/llama_index/blob/2ba13544cd2583418cbeade5bea45ff1da7bb7b8/llama-index-core/llama_index/core/utilities/gemini_utils.py#L15

The key is there 🤔

**Comment by ujjwalk906** [comment]
Works in 0.10.21

**Comment by nicoloboschi** [comment]
@logan-markewich I looked at the problem. @ujjwalk906 was having issue because he wasn't using a llamaindex version containing this fix: https://github.com/run-llama/llama_index/pull/11511. The KeyError "model" is pretty clear now.
 
In fact, after he upgraded to 0.10.21 (with that fix in) things started to work correctly.

Also he fixed his problem WITHOUT reverting the google gen upgrade (also your revert is not even released today).

In conclusion, I will re-create the PR with the upgrade (I did more manual tests and everything looks good on my side)
